### PR TITLE
CXSPA-711: Merge existing migrations - Constructors

### PR DESCRIPTION
**Pull request by plabadie**

### CODE REVIEW

File: tools/breaking-changes/data/breaking-changes.json [hunk 1]
```diff
@@ -93281,6 +93564,26 @@
...
+        "isDeletedMember": true,
+        "deletedMember": {
+          "kind": "Property",
+          "name": "router",
+          "type": "Router"
+        },
+        "apiElementName": "CartTotalsComponent",
+        "apiElementKind": "Class",
+        "entryPoint": "@spartacus/storefront",
+        "deletedComment": "// TODO:Spartacus - Property 'router' was removed from Class 'CartTotalsComponent'.",
```
Comment (bgambocjaviniar) [inline tools/breaking-changes/data/breaking-changes.json:93583]: Just to understand, how did this get added? was this entry added manually. If it is, everything is making sense again to me :P

File: docs/migration/5_0-generated-typescript-changes-doc.md [hunk 2]
```diff
@@ -276,6 +276,38 @@ constructor(
 ```
 
 
+### Constructor changed.
```
Comment (bgambocjaviniar) [inline docs/migration/5_0-generated-typescript-changes-doc.md:279]: ~~not related to this line, but there's a duplicate entry for `ConfiguratorAttributeDropDownComponent`~~

**Review by bgambocjaviniar — CHANGES_REQUESTED** [review]
Should we clean up the constant as well at the end? (the references taken from projects/schematics/src/shared/constants.ts

File: docs/migration/5_0-generated-typescript-changes-doc.md (hunk 2 above)
Comment (bgambocjaviniar) [reply docs/migration/5_0-generated-typescript-changes-doc.md:279]: nvm my mistake :)

File: tools/breaking-changes/data/breaking-changes.json (hunk 1 above)
Comment (bgambocjaviniar) [reply tools/breaking-changes/data/breaking-changes.json:93583]: to add more to this

I recall the script catches optional constructor params, which creates additional migration. How come it doesn't do this anymore as I'm assuming you added entries like carttotals, formserror, quickorderform manually to the breaking changes json

File: docs/migration/5_0.md [hunk 3]
```diff
@@ -413,18 +360,11 @@ The following keys have been added to `configurator-common.ts`:
 
 ### QuickOrderService
 
-- Removed `ProductAdapter` from constructor.
-- Added `ProductSearchConnector` to constructor.
 - The `search` method was removed. Use `searchProducts` instead.
 - The `removeEntry` method was removed. Use `softDeleteEntry` instead.
-
+FormErrorsComponent
```
Comment (bgambocjaviniar) [inline docs/migration/5_0.md]: remove this :P 

I made a similar mistake when searching back and forth from here in vscode https://github.com/SAP/spartacus/pull/15967#discussion_r917013980

File: docs/migration/5_0.md [hunk 4]
```diff
@@ -714,57 +628,7 @@ Due to i18next migration, certain set of keys have been migrated (from `_plural`
 ### OnNavigateFocusService
 
 - This service had been removed in favour of autoscroll fixes made to the `Router`.
-
-### AsmMainUICmmponent
```
Comment (bgambocjaviniar) [inline docs/migration/5_0.md:718]: 😆 and CxAgentAuthService

File: docs/migration/5_0.md [hunk 5]
```diff
@@ -714,57 +628,7 @@ Due to i18next migration, certain set of keys have been migrated (from `_plural`
...
-
-- Added `UserAccountFacade` to constructor.
-- Removed `UserService` from constructor. 
-
-### B2BUserEffects 
-
-- Added `UserAccountFacade` to constructor.
-- Removed `UserService` from constructor.
-  
-### ApproverGuard
```
Comment (bgambocjaviniar) [inline docs/migration/5_0.md:763]: B2BUserEffects and ApproverGuard are missing for migrations

Comment (plabadie) [reply docs/migration/5_0.md:763]: They were not exposed in the public api in 4.3, so no breaking change material for those classes.

File: docs/migration/5_0.md (hunk 3 above)
Comment (plabadie) [reply docs/migration/5_0.md]: Good catch!  Copy paste + search  mistake

File: tools/breaking-changes/data/breaking-changes.json (hunk 1 above)
Comment (plabadie) [reply tools/breaking-changes/data/breaking-changes.json:93583]: When constructor deprecations are not done [as they should](https://sap.github.io/spartacus-docs/breaking-changes/#adding-new-constructor-dependencies-in-minor-versions), with all the constructor signatures above the implementation, MS api extractor doesn't see all signatures.  

Said differently, when we override a constructor, we have to declare all the signatures on top of the implementation.

**Review by bgambocjaviniar — APPROVED** [review]

**Pull request merged**